DelaunayTriangulationSwift/Vertex.swift

```diff
@@ -6,46 +6,45 @@
 //  Copyright © 2016 zero. All rights reserved.
 //
 
+import CoreGraphics
+
 public struct Vertex {
     
     public init(x: Double, y: Double) {
         self.x = x
         self.y = y
     }
     
+    public func pointValue() -> CGPoint {
+        return CGPoint(x: x, y: y)
+    }
+    
     public let x: Double
     public let y: Double
 }
 
-extension Vertex: Equatable {
-    static public func ==(lhs: Vertex, rhs: Vertex) -> Bool {
-        return lhs.x == rhs.x && lhs.y == rhs.y
-    }
-}
+extension Vertex: Equatable { }
 
-extension Vertex: Hashable {
-    public var hashValue: Int {
-        return "\(x)\(y)".hashValue
-    }
+public func ==(lhs: Vertex, rhs: Vertex) -> Bool {
+    return lhs.x == rhs.x && lhs.y == rhs.y
 }
 
-
 extension Array where Element:Equatable {
     func removeDuplicates() -> [Element] {
         var result = [Element]()
-
+        
         for value in self {
             if result.contains(value) == false {
                 result.append(value)
             }
         }
-
+        
         return result
     }
 }
 
-extension Array where Element: Hashable {
-    func removeDuplicates() -> [Element] {
-        return Array(Set(self))
+extension Vertex: Hashable {
+    public var hashValue: Int {
+        return Int((x+y)*10000000)
```

Comment:
you right, I did not think about it in this way :) But I think converting to string and then back is inefficient. Will think about it more.